pulsar-client-cpp/include/pulsar/ReaderConfiguration.h

```diff
@@ -86,6 +86,9 @@ class ReaderConfiguration {
     void setSubscriptionRolePrefix(const std::string& subscriptionRolePrefix);
     const std::string& getSubscriptionRolePrefix() const;
 
+    ReaderConfiguration& setReadCompacted(bool compacted);
```

Comment:
Even though there's a mix of styles in the current config objects, we should try to keep the same return type for the same flag: 

`void ConsumerConfiguration::setReadCompacted(bool compacted);`

vs

` ReaderConfiguration& ReaderConfiguration::setReadCompacted(bool compacted);`

Reply:
Yes, the change is following Ivan's comments, will try to use `void ConsumerConfiguration::setReadCompacted(bool compacted)` to keep alignment. 
And seems it is needed for python side.